enum.rb

```diff
@@ -0,0 +1,124 @@
+# frozen_string_literal: true
+
+# Microverse-202-Advanced-Building-Blocks-Enumerables
+# By Jair Jaramillo
+
+# module enum
+module Enumerable
+  def bubble_sort(array)
+    (array.size - 1).times do |x|
+      array[x], array[x + 1] = array[x + 1], array[x] if array[x] > array[x + 1]
+    end
+    (array.size - 1).times do |x|
+      bubble_sort(array) if array[x] > array[x + 1]
+    end
+    array
+  end
+
+  def bubble_sort_by(arra)
+    (arra.size - 1).times do |x|
+      arra[x], arra[x + 1] = arra[x + 1], arra[x] if yield(arra[x], arra[x + 1]).positive?
+    end
+    (arra.size - 1).times do |x|
+      next unless (arra[x].length - arra[x + 1].length).positive?
+
+      bubble_sort_by(arra) do |_left, _right|
+        yield(arra[x], arra[x + 1])
+      end
+    end
+    arra
+  end
+
+  def my_each(counter = 0)
+    while counter < to_a.length
+      yield to_a[counter]
+      counter += 1
+    end
+  end
+
+  def my_each_with_index(counter = 0)
+    while counter < to_a.length
+      yield to_a[counter], counter
+      counter += 1
+    end
+  end
+
+  def my_select
+    result = []
+    i = 0
+    while i < to_a.length
+      result << to_a[i] if yield to_a[i]
+      i += 1
+    end
+    result
+  end
+
+  def my_all?(pattern = false)
+    if block_given?
+      my_each { |item| return false if !(yield item) == false }
+    elsif !pattern == true
+      my_each { |item| return false if (pattern == item) == false }
+    else
+      my_each { |item| return false unless item == false }
+    end
+    true
+  end
+
+  def my_any?(pattern = false)
+    if block_given?
+      my_each { |item| return true if yield item }
+    elsif !pattern == true
```

Comment:
- [ ] my_any? when no block or argument is given returns true if at least one of the collection is not false or nil
- [ ] my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class
- [ ] my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex
- [ ] my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern